ff/src/fields/mod.rs

```diff
@@ -486,6 +485,31 @@ impl LegendreSymbol {
     }
 }
 
+pub enum SqrtPrecomputation<F: Field> {
```

Comment:
Let's mark this enum as `#[non_exhaustive]`, so that in the future when we add more variants, it won't cause breakage.